tutorials/shaders/compute_shaders.rst

```diff
@@ -0,0 +1,242 @@
+    // Each double has 8 byte (64 bit) so 10 x 8 = 80 bytes
+    var buffer = rd.StorageBufferCreate((uint)inputBytes.Length, inputBytes);
+
+With the buffer in place we need to tell the rendering device to use this buffer.
+To do that we will need to create a uniform (like in normal shaders) and assign it to a uniform set which we can pass to our shader later.
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    # Create a uniform to assign the buffer to the rendering device
+    var uniform := RDUniform.new()
+    uniform.uniform_type = RenderingDevice.UNIFORM_TYPE_STORAGE_BUFFER
+    uniform.binding = 0
+    uniform.add_id(buffer)
+    var uniform_set := rd.uniform_set_create([uniform], shader, 0)
+    
+ .. code-tab:: csharp
+ 
+    // Create a uniform to assign the buffer to the rendering device
+    var uniform = new RDUniform
+    {
+        UniformType = RenderingDevice.UniformType.StorageBuffer,
+        Binding = 0
+    };
+    uniform.AddId(buffer);
+    var uniformSet = rd.UniformSetCreate(new Array<RDUniform> { uniform }, shader, 0);
+
+
+Defining a compute pipeline
+---------------------------
+The next step is to create a range of instructions our GPU can execute.
+We need a pipeline and a compute list for that.
+
+The steps we need to do to compute our result are: 
+
+1. Create a new pipeline.
+2. Begin a list of instructions for our GPU to execute.
+3. Bind our compute list to our pipeline
+4. Bind our buffer uniform to our pipeline
+5. Execute the logic of our shader
+6. End the list of instructions
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    # Create a compute pipeline
+    var pipeline := rd.compute_pipeline_create(shader)
+    var compute_list := rd.compute_list_begin()
+    rd.compute_list_bind_compute_pipeline(compute_list, pipeline)
+    rd.compute_list_bind_uniform_set(compute_list, uniform_set, 0)
+    rd.compute_list_dispatch(compute_list, 5, 1, 1)
+    rd.compute_list_end()
+    
+ .. code-tab:: csharp
+ 
+    // Create a compute pipeline
+    var pipeline = rd.ComputePipelineCreate(shader);
+    var computeList = rd.ComputeListBegin();
+    rd.ComputeListBindComputePipeline(computeList, pipeline);
+    rd.ComputeListBindUniformSet(computeList, uniformSet, 0);
+    rd.ComputeListDispatch(computeList, xGroups: 5, yGroups: 1, zGroups: 1);
+    rd.ComputeListEnd();
+
+
+
+Execute a compute shader
+------------------------
+
+After all of this we are done, kind of.
+We still need to execute our pipeline, everything we did so far was only definition not execution.
+
+To execute our compute shader we just need to submit the pipeline to the GPU and wait for the execution to finish:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    # Submit to GPU and wait for sync
+    rd.submit()
+    rd.sync()
+    
+ .. code-tab:: csharp
+ 
+    // Submit to GPU and wait for sync
+    rd.Submit();
+    rd.Sync();
+
+Congrats you created and executed a compute shader. But wait, where are the results now.
```

Comment:
```suggestion

Ideally, you would not synchronize the RenderingDevice right away as it will cause the CPU to wait for the GPU to finish working. In our example we synchronize right away because we want our data available for reading right away. In general, you will want to wait at least a few frames before synchronizing so that the GPU is able to run in parellel with the CPU. 

Congrats you created and executed a compute shader. But wait, where are the results now.
```